ds-card-reader/driver/driver.go

```diff
@@ -1,47 +1,55 @@
-// Copyright © 2022 Intel Corporation. All rights reserved.
+// Copyright © 2023 Intel Corporation. All rights reserved.
 // SPDX-License-Identifier: BSD-3-Clause
 
 package driver
 
 import (
-	"errors"
 	"fmt"
+	"sync"
 
 	common "ds-card-reader/common"
 	device "ds-card-reader/device"
 
-	dsModels "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
-	"github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
-	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
-	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+	"github.com/edgexfoundry/device-sdk-go/v3/pkg/interfaces"
+	dsModels "github.com/edgexfoundry/device-sdk-go/v3/pkg/models"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v3/models"
 )
 
 // CardReaderDriver represents the EdgeX driver that interfaces with the
 // underlying device
 type CardReaderDriver struct {
 	LoggingClient logger.LoggingClient
+	asyncCh       chan<- *dsModels.AsyncValues
 	CardReader    device.CardReader
 	Config        *device.ServiceConfig
 
-	svc *service.DeviceService
+	sdk interfaces.DeviceServiceSDK
+}
+
+var once sync.Once
+var cardReaderDriver *CardReaderDriver
+
+func NewCardReaderDriver() interfaces.ProtocolDriver {
+	once.Do(func() {
+		cardReaderDriver = new(CardReaderDriver)
+	})
+	return cardReaderDriver
 }
```

Comment:
This is not proper as it use local global vars. NewCardReaderDriver  should simply return  the new cardReaderDriver?

```suggestion
func NewCardReaderDriver() interfaces.ProtocolDriver {
	return &CardReaderDriver{}
}
```